src/cdn.py

```diff
@@ -0,0 +1,31 @@
+import requests, requests_cache
+from cache_requests import install_cache_requests
+import logging
+
+LOG = logging.getLogger(__name__)
+install_cache_requests()
+
+def url_exists(url, msid=None):
+    context = {'msid': msid, 'url': url}
+
+    try:
+        resp = requests.head(url)
+    except requests.ConnectionError:
```

Comment:
it looks like we need three kind of return values
- string, when we have a valid URL
- `None`, when there is no figures PDF
- some invalid value (`False`? placeholder string that is not a URL?), when we encounter a 500 or `ConnectionError` or similar unforeseen condition